sql/alps_submitter.go

```diff
@@ -661,12 +685,71 @@ if __name__ == "__main__":
  					  throttle_steps={{.TrainSpec.throttle_steps}}
 {{end}}
 		),
-		exporter=ArksExporter(export_path=export_path, export_strategy=ExportStrategy.BEST, compare_fn=Closure(best_auc_fn)),
-		model_dir="{{.ScratchDir}}",
-		model_builder=SQLFlowEstimatorBuilder())
+		exporter=ArksExporter(deploy_path=deploy_path, strategy=ExportStrategy.BEST, compare_fn=Closure(best_auc_fn)),
+		runtime=RuntimeConf(
+			model_dir=model_dir,
+		),
+		model_builder=SQLFlowEstimatorBuilder()
+	)
+
+	return experiment
+
+{{else}}
+
+def inference(model_dir):
```

Comment:
It seems that this Go template is becoming longer.

It seems to me we need the following stuff in this Python template:

1. the Estimator builder class
1. the train function
1. the predict function
1. the main function

To make it easy to maintain and test, I would move 1., 2., and 3. out from this .go file into a Python file `python/alps_codegen.py`. So we could have `python/alps_codegen_test.py` for unit testing these facilities. To make this happen, we need to remove all template syntax from 1., 2., and 3. to make them legal Python code. I think this should be pretty easy -- we can change template placeholders into function parameters or class data members.

And, I would leave all template syntax in 4., and keep 4. in this .go file.

Following the above steps, this alps_codegen.go file would generate a short .py file which contains only a main function, which, calls functions/classes defined in python/alps_codegen.py.

To make this solution complete, we should remember that we need to pack python/alps_codegen.py into our release Docker image, so it could be called by the generated Alps submitter programs.

Reply:
Thanks for this comment which is reasonable and valuable, I will close this PR and do the refactoring of the code template at first.